### PR TITLE
store/tikv: move kv.TxnInfo to store/tikv

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
`kv.TxnInfo` is defined in `kv` but mainly used in `store/tikv`. In fact, SQL layer should not depend on specific storage engine and does not really care about what's inside the info so we can use string to replace the struct.

Part of https://github.com/pingcap/tidb/issues/22513

### What is changed and how it works?
What's Changed:
- move `TxnInfo` to `store/tikv`
- return string to SQL layer
- update test

### Check List 
Tests 
- Unit test

Side effects
- Breaking backward compatibility
When `txnInfo` is not set. Previously `select @@tidb_last_txn_info` returns an empty json string respects tikv txn info structure.
After this PR it will return empty string "".

### Release note 
- No release note

### CODE REVIEW

**Review by AndreMouche — APPROVED** [review]
LGTM

File: store/tikv/txn.go [hunk 1]
```diff
@@ -393,7 +405,9 @@ func (txn *KVTxn) onCommitted(err error) {
 		if err != nil {
 			info.ErrMsg = err.Error()
 		}
-		txn.commitCallback(info, err)
+		infoStr, err2 := json.Marshal(info)
+		_ = err2
```
Comment (AndreMouche) [inline store/tikv/txn.go:409]: Does it seem we need to keep the err2? the related test case is `TestTiDBLastTxnInfo`

**Review by AndreMouche — APPROVED** [review]
The test `TestTiDBLastTxnInfo` failed because we try to json.unmarshal an empty string

**Review by AndreMouche — CHANGES_REQUESTED** [review]
The test `TestTiDBLastTxnInfo` failed because we try to json.unmarshal an empty string

**Comment by disksing** [comment]
> The test `TestTiDBLastTxnInfo` failed because we try to json.unmarshal an empty string

I'll take a look.

**Review by AndreMouche — APPROVED** [review]
LGTM

File: store/tikv/txn.go (hunk 1 above)
Comment (disksing) [reply store/tikv/txn.go:409]: The test is not related to this. err2 should always be nil.

**Comment by lysu** [comment]
this function only be used in automated-tests https://github.com/pingcap/automated-tests/search?q=tidb_last_txn_info
and "return empty string" doesn't break test case, so IMO it's ok

/LGTM

File: sessionctx/variable/varsutil.go [hunk 2]
```diff
@@ -178,11 +178,7 @@ func GetSessionOnlySysVars(s *SessionVars, key string) (string, bool, error) {
 	case TiDBCurrentTS:
 		return fmt.Sprintf("%d", s.TxnCtx.StartTS), true, nil
 	case TiDBLastTxnInfo:
-		info, err := json.Marshal(s.LastTxnInfo)
-		if err != nil {
-			return "", true, err
-		}
-		return string(info), true, nil
+		return s.LastTxnInfo, true, nil
```
Comment (lysu) [inline sessionctx/variable/varsutil.go:181]: or maybe we can

```
if s.LastTxnInfo ==  "" {
    return "{}"
}
```

if we want to keep compatible to old one but it does required 🐱

**Comment by zyguan** [comment]
> this function only be used in automated-tests https://github.com/pingcap/automated-tests/search?q=tidb_last_txn_info
> and "return empty string" doesn't break test case, so IMO it's ok, cc @zyguan

Jepsen also use this variable [here](https://github.com/pingcap/jepsen/blob/testing_ci/tidb/src/tidb/sql.clj#L299). But never mind, I will update these functions later.

**Comment by disksing** [comment]
Thanks @zyguan , I'll merge it then.

**Pull request merged**